### ISSUE TITLE
Save PDF Background in the XOPP file instead of accesing an external pdf file

### ISSUE THREAD

**Issue by CreativerName** [xournalpp/xournalpp#4882]
I have the Problem that when I want so take my xopp files from one PC to another, all the PDF background aren't there anymore because the pdf with my background has an new location because you are on a new PC and I have to manually re-link the Xournal files to my PDF-backgrounds . And additionally the pdfs are additional files which lay around in my folder and that's triggering me.

My Idea is that you implement that you can add PDF or picture backgrounds and these backgrouds are saved within the xopp file. In my case that would make my Xournal experience way more comfortable as I can open the Xournal files from any devices and I don't have additional files in my Folder.

I hope the feature is implementable, but for now thaks for at least reading my idea.

**Comment by sebiik** [comment]
Including the PDF in the xopp file is not a good solution IMHO.
- Some PDFs are huge.
- I often find myself opening the PDF in a generic PDF reader (to keep a certain page open, or to search for text) and the Xopp file in Xournal to make notes.
- If you include the PDF in the Xopp you lose a lot of flexibility and it may even lead to licensing problems, depending on where the PDF came from.

A more elegant solution would be to either
- have Xournal++ check the absolute path of the PDF just as it currently does, and if it does not find it there, do another search in the working directory of the .xopp file. This would be the best of two worlds IMHO, as you don't have to search for and update the PDF path every time.
- allow for relative paths, e.g. "myfile.pdf" same as "./myfile.pdf", "../pdf/myfile.pdf" etc.

**Comment by fastpinguin** [comment]
I do not agree with @sebiik. I used xournalpp over a longer period of time and have more than once deleted pdfs that I used as backgrounds while reorganizing (pretty painstaking to redo that ...). If one file will not work without the other, then they should be one file.

* Regarding huge pdfs. I do not perceive any problem regarding this. I have to save the pdf anyway, so lets rather put it in a place that the user can not accidentally delete. There is no storage space saved by using a two-file-approach, as the original can be discarded instead.
* Opening the pdf. Nothing would prevent you from keeping a second copy for this purpose. However, with the limited screenspace on devices that xournalpp is most likely to be used, e.g. Microsoft Surface Tablets..., I would much rather scroll around in the document than have a second window open - but this is subjective, i know
* Flexibilty, Licensing problems: Nothing would prevent the implementation to include a "discard pdf background" in the export tab.

I agree that xournalpp should also be more flexible regarding pdf paths. Nevertheless, including pdfs in xopp files should be a priority IMHO.
